eslint/eslint-patch/src/eslint-bulk-suppressions/bulk-suppressions-patch.ts

```diff
@@ -0,0 +1,295 @@
+
+  if (
+    !json.suppressions.every((suppression) => {
+      if (typeof suppression !== 'object')
+        throw new Error(`Invalid suppression: ${JSON.stringify(suppression, null, 2)}`);
+      if (suppression === null)
+        throw new Error(`Suppression is null: ${JSON.stringify(suppression, null, 2)}`);
+      if (!suppression.hasOwnProperty('file'))
+        throw new Error(`Missing "file" property in suppression: ${JSON.stringify(suppression, null, 2)}`);
+      if (typeof suppression.file !== 'string')
+        throw new Error(
+          `"file" property in suppression is not a string: ${JSON.stringify(suppression, null, 2)}`
+        );
+      if (!suppression.hasOwnProperty('scopeId'))
+        throw new Error(`Missing "scopeId" property in suppression: ${JSON.stringify(suppression, null, 2)}`);
+      if (typeof suppression.scopeId !== 'string')
+        throw new Error(
+          `"scopeId" property in suppression is not a string: ${JSON.stringify(suppression, null, 2)}`
+        );
+      if (!suppression.hasOwnProperty('rule'))
+        throw new Error(`Missing "rule" property in suppression: ${JSON.stringify(suppression, null, 2)}`);
+      if (typeof suppression.rule !== 'string')
+        throw new Error(
+          `"rule" property in suppression is not a string: ${JSON.stringify(suppression, null, 2)}`
+        );
+      return true;
+    })
+  ) {
+    throw new Error(
+      `Invalid suppression in "suppressions" array: ${JSON.stringify(json.suppressions, null, 2)}`
+    );
+  }
+  return true;
+}
+
+function readSuppressionsJson(fileAbsolutePath: string): BulkSuppressionsJson {
+  const eslintrcDirectory = findEslintrcDirectory(fileAbsolutePath);
+  const suppressionsPath = path.join(eslintrcDirectory, '.eslint-bulk-suppressions.json');
+  let suppressionsJson = { suppressions: [] };
+  try {
+    const fileContent = fs.readFileSync(suppressionsPath, 'utf-8');
+    suppressionsJson = JSON.parse(fileContent);
+
+    if (!validateSuppressionsJson(suppressionsJson)) {
+      console.warn(
+        `Unexpected file content in .eslint-bulk-suppressions.json. JSON expected to be in the following format:
+{
+  suppressions: {
+      file: string;
+      scopeId: string;
+      rule: string;
+  }[];
+}
+Please check file content, or delete file if suppressions are no longer needed.
+`
+      );
+      suppressionsJson = { suppressions: [] };
+    }
+  } catch {
+    // Do nothing and let JSON.parse() log the error. suppressionsJson will stay as the initialized value.
+  }
+  return suppressionsJson;
+}
+
+function shouldWriteSuppression(fileAbsolutePath: string, suppression: Suppression): boolean {
+  if (process.env.ESLINT_BULK_SUPPRESS === undefined) return false;
+
+  if (isSuppressed(fileAbsolutePath, suppression)) return false;
+
+  const rulesToSuppress = process.env.ESLINT_BULK_SUPPRESS.split(',');
+
+  if (rulesToSuppress.length === 1 && rulesToSuppress[0] === '*') return true;
```

Comment:
The repository's lint ruleset should be banning this formatting; we require `{}` braces for control flow code blocks.

Reply:
@dmichon-msft  ESLint is disabled for this project.  If I remember right, it was due to this problem:

https://github.com/microsoft/rushstack/blob/32536c208fd257aaf0a9302bbeadd26b210a9f7f/eslint/eslint-plugin-packlets/.eslintrc.js.disabled#L1-L5

Since we are adding more code to these projects, now might be a good time to revisit, and see if we can get the lint rules enabled for these projects.

I did already enable ESLint for his new `eslint-bulk` project, but fixing preexisting issues should be out of scope for this PR.  So I agree we should fix it, but it can be addressed in a separate PR instead of holding up the release of @kevin-y-ang's feature.